src/components/Dropdown.js

```diff
@@ -0,0 +1,53 @@
+export class Dropdown extends HTMLElement {
+  _open = false;
+  _button;
+  _expandableBox;
+
+  openCloseMenuHandler = () => {
+    this._open = !this._open;
+    this.updateRender();
+  };
+
+  constructor() {
+    super();
+    this._button = this.querySelector("dp-dropdown>dp-dropbdown-button");
+    this._expandableBox = this.querySelector("dp-dropdown>dp-dropdown-content");
+
+    //TODO: use class for hide or show expandableBox
+    this._expandableBox.setAttribute(
+      "style",
+      `display: ${this._open ? "block" : "none"}`
+    );
+
+    this.append(this._button, this._expandableBox);
+  }
+
+  updateRender() {
+    //TODO: use class for hide or show expandableBox
+    this._expandableBox.setAttribute(
+      "style",
+      `display: ${this._open ? "block" : "none"}`
+    );
+  }
+
+  collapse() {
+    this._open = false;
+    this.updateRender();
+  }
+
+  _collapseOther() {
+    const all = document.querySelectorAll("dp-dropdown");
+    all.forEach((dropdown) => {
+      if (dropdown !== this) {
+        dropdown.collapse();
+      }
+    });
+  }
```

Comment:
It is nice, but I think that it will not work with legacy styles, maybe you should add a fallback like:

```suggestion
  _collapseOther() {
    const all = document.querySelectorAll("dp-dropdown");
    all.forEach((dropdown) => {
      if (dropdown !== this) {
        if (dropdown.collapse) {
          dropdown.collapse();
        } else {
          // TODO: fallback
        }
      }
    });
  }
```